scripts/solve_network.py

```diff
@@ -314,78 +319,72 @@ def add_chp_constraints(n):
     electric_fix = n.links.index[electric_bool & ~n.links.p_nom_extendable]
     heat_fix = n.links.index[heat_bool & ~n.links.p_nom_extendable]
 
-    link_p = get_var(n, "Link", "p")
+    link_p = n.model["Link-p"]
 
     if not electric_ext.empty:
-        link_p_nom = get_var(n, "Link", "p_nom")
+        link_p_nom = n.model["Link-p_nom"]
 
         # ratio of output heat to electricity set by p_nom_ratio
-        lhs = linexpr(
-            (
-                n.links.loc[electric_ext, "efficiency"]
-                * n.links.loc[electric_ext, "p_nom_ratio"],
-                link_p_nom[electric_ext],
-            ),
-            (-n.links.loc[heat_ext, "efficiency"].values, link_p_nom[heat_ext].values),
-        )
+        lhs = (
+            n.links.loc[electric_ext, "efficiency"]
+            * n.links.loc[electric_ext, "p_nom_ratio"]
+            * link_p_nom.loc[electric_ext]
+        ) + (-n.links.loc[heat_ext, "efficiency"] * link_p_nom.loc[heat_ext])
 
-        define_constraints(n, lhs, "=", 0, "chplink", "fix_p_nom_ratio")
+        n.model.add_constraints(lhs == 0, name="chplink-fix_p_nom_ratio")
 
         # top_iso_fuel_line for extendable
-        lhs = linexpr(
-            (1, link_p[heat_ext]),
-            (1, link_p[electric_ext].values),
-            (-1, link_p_nom[electric_ext].values),
+        lhs = (
+            (1 * link_p.loc[heat_ext]),
+            (1 * link_p.loc[electric_ext]),
+            (1 * link_p_nom.loc[electric_ext]),
         )

```

Comment:
This one cannot work.
It feels to me that probably we need to slightly change the CI options to enable more of these functions to actually be tested.
Moreover, pay attention on signs, it seems something has went wrong here

Reply:
Thanks @davide-f for catching this. I will revise the implementation.